src/metabase/notification/payload/impl/dashboard.clj

```diff
@@ -56,8 +55,9 @@
   [{:keys [id creator_id handlers] :as notification-info} notification-payload]
   ;; clean up all the temp files that we created for this notification
   (try
-    (run! #(when-let [rows (get-in % [:result :data :rows])]
-             (notification.temp-storage/cleanup! rows))
+    (run! #(let [rows (get-in % [:result :data :rows])]
+             (when (and rows (notification.payload/is-cleanable? rows))
+               (notification.payload/cleanup! rows)))
```

Comment:
Maybe it would be nicer to just extend the protocol to do nothing if it's not cleanable? 😁

Reply:
awesome! were quite unhappy with that check. [4400f01](https://github.com/metabase/metabase/pull/55896/commits/4400f01b5eba4d3308689379ac34f40494587be9)